pkgs/applications/blockchains/solana/default.nix

```diff
@@ -59,6 +46,21 @@ rustPlatform.buildRustPackage rec {
     inherit sha256;
   };
 
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
you can also use "${src.outPath}/Cargo.lock" to avoid having to update the cargo lock in this repo

Reply:
Yes but the [documentation](https://ryantm.github.io/nixpkgs/languages-frameworks/rust/) told me to do this way

Reply:
outcome is the same, only benefit is it saves you from keeping the cargo.lock in this repo and updating it the next time version is bumped. Unsure where I saw this pattern. Unfortunately I'm not that versed in the Rust ecosystem

Reply:
Some packages did it so I did the same, also makes the person updating needing to see the lockfile, or at least is what I pray happens

Reply:
@arijoon the pattern you are mentioning is interesting for personal repos, but doesn't work for nixpkgs. it uses Import From Derivation which is disabled in nixpkgs. You can try it, but ofborg won't allow it.

Reply:
> @arijoon the pattern you are mentioning is interesting for personal repos, but doesn't work for nixpkgs. it uses Import From Derivation which is disabled in nixpkgs. You can try it, but ofborg won't allow it.

I didn't know that, thanks for pointing out